nucliadb_paragraphs/src/stop_words.rs

```diff
@@ -0,0 +1,116 @@
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::fs;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+
+use lazy_static::lazy_static;
+use lingua::Language;
+use lingua::Language::English;
+use lingua::{LanguageDetector, LanguageDetectorBuilder};
+use serde_json;
+
+lazy_static! {
+    static ref LANGUAGE_DETECTOR: LanguageDetector = {
+        let detector = LanguageDetectorBuilder::from_all_spoken_languages()
+            .with_preloaded_language_models()
+            .build();
+        detector
+    };
+}
+
+struct StopWords {
+    map: HashMap<String, HashSet<String>>,
+}
+
+impl StopWords {
+    fn new() -> StopWords {
+        StopWords {
+            map: HashMap::new(),
+        }
+    }
+
+    fn load_language(&mut self, language_code: &str) -> Result<(), std::io::Error> {
+        let file_path = format!("stop_words/{}.json", language_code);
+        let file = File::open(file_path)?;
+        let reader = BufReader::new(file);
+        let stop_words: HashSet<String> = serde_json::from_reader(reader)?;
+        self.map.insert(language_code.to_string(), stop_words);
+        Ok(())
+    }
+
+    fn is_stop_word(&self, word: &str, language_code: &str) -> bool {
+        if let Some(stop_words) = self.map.get(language_code) {
+            return stop_words.contains(word);
+        }
+
+        false // Language not found or lazy_static not initialized
+    }
+}
+
+lazy_static! {
+    static ref STOP_WORDS: StopWords = {
+        let mut stop_words = StopWords::new();
+
+        let json_dir = Path::new("stop_words");
+        if let Ok(entries) = fs::read_dir(json_dir) {
+            for entry in entries {
+                if let Ok(entry) = entry {
+                    if let Some(extension) = entry.path().extension() {
+                        if extension == "json" {
+                            if let Some(language_code) = entry.path().file_stem() {
+                                if let Some(language_code) = language_code.to_str() {
+                                    if let Err(err) = stop_words.load_language(language_code) {
+                                        eprintln!(
+                                            "Error loading stop words for {}: {}",
+                                            language_code, err
+                                        );
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        stop_words
+    };
+}
+
+pub fn detect_language(query: &str) -> Language {
+    let detected_language: Option<Language> = LANGUAGE_DETECTOR.detect_language_of(query);
+
+    let res = detected_language.unwrap_or(English);
+    println!("Detected language for {:?} is {:?}", query, res);
```

Comment:
here too

Reply:
yeah that was for debugging, the final patch won't have them ;)